src/scenes/Question/scenes/Answer/Answer.jsx

```diff
@@ -46,19 +52,27 @@ class Answer extends Component {
     }
   }
 
-  cleanSources() {
-    const { sources } = this.state
-    return sources
-      .map(s => {
-        if (s.new) return omit(s, ['id', 'new'])
-        return s
+  sourcesToList(sources) {
+    return sources.map(({ id, label, url }) => ({
+      id,
+      key: label,
+      value: url
+    }))
+  }
+
+  listToSources(list) {
+    return list
+      .map(({ id, key, value }) => {
+        if (isUuidV4(id)) {
+          return { label: key, url: value }
+        }
+        return { id, label: key, url: value }
       })
       .filter(s => s.label !== '' && s.url !== '')
   }
 
   onTextChange = value => this.setState({ answer: value })
-
-  onSourceChange = sources => this.setState({ sources: sources })
+  onSourcesChange = onListChange(this.setState.bind(this), 'sources')
```

Comment:
An arrow instead of bind would be clearer I think

Reply:
I don't have a strong opinion about this. We could use `state => this.setState(state)`, but we would lose the second argument of setState (`callback`). We would then need to use: `(...args) => this.setState(...args)` which I find repetitive. Having this.setState.bind(this) is more "correct" in what we want to do here: pass the `setState` function to be used as is.

Reply:
OK then.